qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -104,24 +104,54 @@ class OptimizationResult:
         results: The original results object returned from the optimization algorithm. This can
             contain more information than only the optimal value and function value.
         status: The termination status of the algorithm.
+        variables: The list of variables under optimization.
     """
 
     Status = OptimizationResultStatus
 
     def __init__(self, x: Optional[Any] = None, fval: Optional[Any] = None,
                  results: Optional[Any] = None,
-                 status: OptimizationResultStatus = OptimizationResultStatus.SUCCESS) -> None:
-        self._val = x
+                 status: OptimizationResultStatus = OptimizationResultStatus.SUCCESS,
+                 variables: Optional[List[Variable]] = None) -> None:
+        self._x = x  # pylint: disable=invalid-name
```

Comment:
May be we should have variable names as a mandatory parameter of the constructor? Names are always present in QP, why don't keep them from the problem by default.

Reply:
I agree. It would be better to have variables names as mandatory parameters.

By the way, I think other params, e.g., x and fval, would be better to be mandatory. It will be another issue to address.

Reply:
Yes, agree, `x` and `fval` should be mandatory with an option to set them to `None`.

Reply:
since variable names are contained in 'variables', I would not bloat the constructor params

Reply:
thanks @adekusar-drl 
I refactored the code based on your comments

Reply:
> I agree. It would be better to have variables names as mandatory parameters.
> 
> By the way, I think other params, e.g., x and fval, would be better to be mandatory. It will be another issue to address.

class LinearEqualityToPenalty is using OptimizationResult() with empty constructor

Reply:
This bunch of lines can be converted to accommodate mandatory parameters.

```python
        new_result = OptimizationResult()
        new_result.x = result.x

        # Set the new function value
        new_result.fval = substituted_qp.objective.constant

        # Set the new status of optimization result
        if substituted_qp.status == QuadraticProgramStatus.VALID:
            new_result.status = OptimizationResultStatus.SUCCESS
        else:
            new_result.status = OptimizationResultStatus.INFEASIBLE

        return new_result
```